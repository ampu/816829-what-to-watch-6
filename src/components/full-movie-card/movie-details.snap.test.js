import React from 'react';

import {render} from '@testing-library/react';

import MovieDetails from './movie-details';

describe(`MovieDetails component`, () => {
  it(`Should render according to the snapshot with no movie provided`, () => {
    const {container} = render(<MovieDetails/>);
    expect(container).toMatchSnapshot();
  });

  it(`Should render according to the snapshot with movie provided`, () => {
    const movie = {
      id: `fakeId`,
      director: `fakeDirector`,
      stars: [`fakeStarOne`, `fakeStarTwo`],
      durationSeconds: 4255,
      genre: `fakeGenre`,
      year: 2021,
    };
    const {container} = render(<MovieDetails movie={movie}/>);
    expect(container).toMatchSnapshot();
  });
});
