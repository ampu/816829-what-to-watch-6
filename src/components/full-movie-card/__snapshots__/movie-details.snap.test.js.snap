// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails component Should render according to the snapshot with movie provided 1`] = `
<div>
  <div
    class="movie-card__text movie-card__row"
  >
    <div
      class="movie-card__text-col"
    >
      <p
        class="movie-card__details-item"
      >
        <strong
          class="movie-card__details-name"
        >
          Director
        </strong>
        <span
          class="movie-card__details-value"
        >
          fakeDirector
        </span>
      </p>
      <p
        class="movie-card__details-item"
      >
        <strong
          class="movie-card__details-name"
        >
          Starring
        </strong>
        <span
          class="movie-card__details-value movie-card__details-value--stars"
        >
          fakeStarOne,
fakeStarTwo
        </span>
      </p>
    </div>
    <div
      class="movie-card__text-col"
    >
      <p
        class="movie-card__details-item"
      >
        <strong
          class="movie-card__details-name"
        >
          Run Time
        </strong>
        <span
          class="movie-card__details-value"
        >
          1h 10m
        </span>
      </p>
      <p
        class="movie-card__details-item"
      >
        <strong
          class="movie-card__details-name"
        >
          Genre
        </strong>
        <span
          class="movie-card__details-value"
        >
          fakeGenre
        </span>
      </p>
      <p
        class="movie-card__details-item"
      >
        <strong
          class="movie-card__details-name"
        >
          Released
        </strong>
        <span
          class="movie-card__details-value"
        >
          2021
        </span>
      </p>
    </div>
  </div>
</div>
`;

exports[`MovieDetails component Should render according to the snapshot with no movie provided 1`] = `
<div>
  <div
    class="movie-card__text movie-card__row"
  >
    <div
      class="movie-card__text-col"
    >
      
    </div>
    <div
      class="movie-card__text-col"
    >
      
    </div>
  </div>
</div>
`;
